compat/mingw.c

```diff
@@ -3296,8 +3587,32 @@ static void setup_windows_environment(void)
 			struct strbuf buf = STRBUF_INIT;
 			strbuf_addstr(&buf, tmp);
 			if ((tmp = getenv("HOMEPATH"))) {
+				/*
+				 * Since Vista, HOMEDRIVE/HOMEPATH are set to Windows\System32
+				 * for secondary logons/services instead of being unset.
+				 * We don't want to (and likely can't) save stuff there
+				 * so ignore them in such a case.
+				 */
+				wchar_t w_system_dir[MAX_PATH];
+				wchar_t *w_buf;
 				strbuf_addstr(&buf, tmp);
-				if (is_directory(buf.buf))
+				{
+					UINT len_system_dir;
+					size_t len_w_buf;
+					if (!(len_system_dir = GetSystemDirectoryW(
+						      (LPWSTR)w_system_dir,
+						      ARRAY_SIZE(w_system_dir))))
+						die("GetSystemDirectoryW failed: 0x%08lx",
+						    GetLastError());
```

Comment:
Do we have any precedent of just coughing up a raw hexadecimal Windows System error?

Reply:
Yes -- e.g. at `WSAStartup` in this file and in `git_CC_error_check` uses elsewhere.
I can't imagine why this API would fail other than insufficient buffer -- and `MAX_PATH` is more than sufficient. So if it fails, something is seriously wrong and we cannot do anything meaningful about it anyway.

And Windows NTSTATUS error codes are more readable in hex than decimal...
Oh, I missed that this is not an NTSTATUS. Yes, a decimal would be more logical then.

Reply:
I was thinking along the lines of `err_win_to_posix()` & `die_errno()` or maybe even `FormatMessage()`. Thinking about it some more, should we `die()` here at all? We failed to get thesystem dir, but either 

* the `HOMEDRIVE`/`HOMEPATH` combination points to it and the user gets the #2709 behaviour, which is inconvenient, but not completely unusable or
* it points to a sensible `HOME` directory, in which case we really shouldn't just quit.

Maybe a `warning()` would be better? Or  maybe we should just silently accept that we can't determine the system directory?